WordPress/Classes/Models/Blog+Post.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+public extension Blog {
+    /// Lookup a post in the blog.
+    ///
+    /// - Parameter postID: The ID associated with the post.
+    /// - Returns: The `AbstractPost` associated with the given post ID.
+    @objc(lookupPostWithID:inContext:)
+    func lookupPost(withID postID: NSNumber, in context: NSManagedObjectContext) -> AbstractPost? {
+        lookupPost(withID: postID.int64Value, in: context)
+    }
+
+    /// Lookup a post in the blog.
+    ///
+    /// - Parameter postID: The ID associated with the post.
+    /// - Returns: The `AbstractPost` associated with the given post ID.
+    func lookupPost(withID postID: Int64, in context: NSManagedObjectContext) -> AbstractPost? {
+        let request = NSFetchRequest<AbstractPost>(entityName: NSStringFromClass(AbstractPost.self))
+        request.predicate = NSPredicate(format: "blog = %@ AND original = NULL AND postID = %@", self, NSNumber(value: postID))
```

Comment:
I noticed we wrap the `Int64` `postId` into an `NSNumber` here and that the method above takes `NSNumber` as input, extracts its `int64Value` and calls this one.

How do you feel about moving the implementation in the method with `NSNumber` input, and make this methods create an `NSNumber` from an `Int64` to call it?

Reply:
I wanted to keep the original predicate. But I probably was a bit too cautious. I've updated this implementation in https://github.com/wordpress-mobile/WordPress-iOS/pull/19317/commits/b7bb788bc07c9533ab4497fa019626dce956555f, and the `NSNumber` back and forth conversion is no longer needed.